### PR TITLE
Count of Digits

### PR DESCRIPTION
**Pull request by tk315**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Count of digits article added

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#81

### CODE REVIEW

**Comment by tk315** [comment]
Please review @ankitaggarwal23 @HarshCasper @iamrajiv

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@tk315 Resolve the merge conflict.

**Comment by tk315** [comment]
@iamrajiv done

File: docs/mathematics/count-of-digits.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+slug: /mathematics/count-of-digits
+---
+
+## Introduction
+
+Count of digits refers to the sum of the frequency of all distinct digits in a number.
+
+## Explanation 
+
+We will make a cnt variable and initilize it to `0`. We will keep on dividing the number by `10` and keep on increasing the cnt variable until the number becomes `0`.
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:12]: Explanation shouldn't be same as algorithm.

File: docs/mathematics/count-of-digits.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+num = 45/10=4       
+cnt = 3
+ 
+num = 4/10=0        
+cnt = 4
+```
+
+## Algorithm
+
+* We will make a cnt variable and initilize it to `0`.
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:34]:
```suggestion
* We will make a `cnt` variable and initialize it to `0`.
```

File: docs/mathematics/count-of-digits.mdx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+* We will keep on dividing the number by `10` and increase the cnt variable until the number becomes `0`.
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:36]:
```suggestion
* We will keep on dividing the number by `10` and increase the `cnt` variable until the number becomes `0`.
```

File: docs/mathematics/count-of-digits.mdx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int count_of_digits(int num)
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:56]: Include a condition when num = 0.

File: docs/mathematics/count-of-digits.mdx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+The count of digits in the given number is: 4
+```
+
+## Complexity Analysis
+
+`n` is the count of digits in the number.
+
+- Time Complexity:
+
+  - **Best Case**: O (n)
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:127]:
```suggestion
  - **Best Case**: `O(n)`
```

File: docs/mathematics/count-of-digits.mdx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+  - **Average Case**: O (n)
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:128]:
```suggestion
  - **Average Case**: `O(n)`
```

File: docs/mathematics/count-of-digits.mdx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  - **Worst Case**: O (n)
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:129]:
```suggestion
  - **Worst Case**: `O(n)`
```

File: docs/mathematics/count-of-digits.mdx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+- Space Complexity: O (1)
```
Comment (atarax665) [inline docs/mathematics/count-of-digits.mdx:131]:
```suggestion
- Space Complexity: `O(1)`
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: docs/mathematics/count-of-digits.mdx (hunk 1 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

File: docs/mathematics/count-of-digits.mdx (hunk 2 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

File: docs/mathematics/count-of-digits.mdx (hunk 3 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

File: docs/mathematics/count-of-digits.mdx (hunk 4 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx:56]: done

File: docs/mathematics/count-of-digits.mdx (hunk 5 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

File: docs/mathematics/count-of-digits.mdx (hunk 6 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

File: docs/mathematics/count-of-digits.mdx (hunk 7 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

File: docs/mathematics/count-of-digits.mdx (hunk 8 above)
Comment (tk315) [reply docs/mathematics/count-of-digits.mdx]: done

**Comment by tk315** [comment]
Please review @ankitaggarwal23 @HarshCasper @iamrajiv @atarax665

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make the same changes I asked here: https://github.com/TesseractCoding/NeoAlgo-Docs/pull/176

**Comment by tk315** [comment]
Please review  @HarshCasper @ankitaggarwal23 @iamrajiv

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**